File: Codeforces_Questions/DIV2_ROUND1011/a.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int t;
    cin >> t;
    while (t--)
    {
        int n, k;
        cin >> n >> k;
        string s;
        cin >> s;

        if (s[0] < s[n - 1])
        {
            cout << "YES" << endl;
            continue;
        }

        if (n == 1 || k == 0)
        {
            cout << "NO" << endl;
            continue;
        }

        bool found = false;
        for (int i = 0; i < n - 1; i++)
        {
            if (s[i] < s[n - 1] || s[i] > s[n - 1])
            {
                cout << "YES" << endl;
                found = true;
                break;
            }
        }

        if (!found)
        {
            cout << "NO" << endl;
        }
    }
    return 0;
}

File: Codeforces_Questions/DIV2_ROUND1011/a.py
t = int(input(""))
while t>0:
    n,k = input("").split(" ")
    n = int(n)
    k = int(k)
    
    s = input("")
    
    if s[0]<s[n-1]:
        print("YES")
        continue
    
    if n==1 or k==0:
        print("NO")
        continue
    
    found = False
    
    for i in s[:n-1]:
        if i>s[n-1] or i<s[n-1]:
            print("YES")
            found = True
            break
    
    if not found:
        print("NO")
            
    t-=1